Subject: [PATCH] git-instaweb: enable source highlighting if available

If the correct highlight executable is available, then configure
gitweb to use it for highlighting the source code.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 git-instaweb.sh | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/git-instaweb.sh b/git-instaweb.sh
index 4c0af04f..a8542fb2 100755
--- a/git-instaweb.sh
+++ b/git-instaweb.sh
@@ -600,6 +600,13 @@ our \$projects_list = \$projectroot;
 
 \$feature{'remote_heads'}{'default'} = [1];
 EOF
+	hlbin="$(command -v highlight 2>/dev/null)"
+	if [ -x "$hlbin" ] && "$hlbin" --version | grep -q 'Andre Simon'; then
+		cat >> "$fqgitdir/gitweb/gitweb_config.perl" <<EOF
+our \$highlight_bin = "$hlbin";
+\$feature{'highlight'}{'default'} = [1];
+EOF
+	fi
 }
 
 configure_httpd() {
---
